src/st/dre/161DC.c

```diff
@@ -139,7 +141,100 @@ void EntityCSMoveAlucard(Entity* self) {
     }
 }
 
-// Delete this once the below entity is decompiled!
-const s32 rodata_pad_118F0 = 0;
 // appears to load from the CD and freeze the game
-INCLUDE_ASM("st/dre/nonmatchings/161DC", EntityUnkId23);
+void EntityUnkId23(Entity* self) {
+    Entity* player = &PLAYER;
+    Entity* ent = &g_Entities[80];
+    s16 diff;
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(D_8018047C);
+        D_8003C8B8 = 0;
+        *D_80097400 = 1;
+        g_Player.padSim = 0;
+        if (g_Player.unk0C & 1) {
```

Comment:
```suggestion
        if (g_Player.unk0C & PLAYER_STATUS_BAT_FORM) {
```
and similar for the lines below